From c0e8e603ef8b65ac039a2699554025661a3f62d2 Mon Sep 17 00:00:00 2001
From: Adrian Negreanu <groleo@gmail.com>
Date: Mon, 23 Jan 2017 12:27:56 +0200
Subject: [PATCH 10/31] cmake: use glws_nonx for Vivante

The nonX windowing system gets used when Apitrace is compiled

with:
   EGL

and without:
   X11
   Win32
   Apple
   Waffle
   Android
---
 retrace/CMakeLists.txt | 28 ++++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/retrace/CMakeLists.txt b/retrace/CMakeLists.txt
index cce69e1..9973e2c 100644
--- a/retrace/CMakeLists.txt
+++ b/retrace/CMakeLists.txt
@@ -158,6 +158,34 @@ if (ENABLE_EGL AND X11_FOUND AND NOT WIN32 AND NOT APPLE AND NOT ENABLE_WAFFLE)
     install (TARGETS eglretrace RUNTIME DESTINATION bin) 
 endif ()
 
+if (Vivante_FOUND AND ENABLE_EGL AND NOT X11_FOUND AND NOT WIN32 AND NOT APPLE AND NOT ENABLE_WAFFLE AND NOT ANDROID)
+    add_executable (eglretrace
+        glws_nonx.cpp
+    )
+
+    add_dependencies (eglretrace glproc)
+    include_directories (${CMAKE_SYSROOT}/usr/include/HAL/)
+
+    target_link_libraries (eglretrace
+        retrace_common
+        glretrace_common
+        glhelpers
+        glproc_egl
+        ${CMAKE_THREAD_LIBS_INIT}
+        ${CMAKE_DL_LIBS}
+        VDK
+    )
+
+    if (${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+        target_link_libraries (eglretrace rt)
+        if (READPROC_H_FOUND)
+            target_link_libraries (eglretrace ${proc_LIBRARY})
+        endif ()
+    endif ()
+
+    install (TARGETS eglretrace RUNTIME DESTINATION bin)
+endif ()
+
 if (ENABLE_EGL)
     if (ENABLE_WAFFLE)
         add_executable (eglretrace
-- 
2.7.4

